front/src/components/ThreeDModel/FurnitureViewer.tsx

```diff
@@ -0,0 +1,145 @@
+  )
+}
+
+const FurnitureScene = memo(function FurnitureScene({
+  width,
+  height,
+  depth,
+  currentPlintHeight,
+  selectedColor,
+}: {
+  width: number
+  height: number
+  depth: number
+  currentPlintHeight: number
+  selectedColor: string
+}) {
+  return (
+    <>
+      <SceneLights enableShadows={true} ambientLightIntensity={0.25} shadowQuality="high" />
+
+      {/* Load background & shadow man GLB models */}
+      <Suspense fallback={null}>
+        <GLBModel
+          modelUrl="/assets/3d-models/bg.glb"
+          modelPosition={[0, 0, 0]}
+          modelScale={[100, 45, 45]}
+          shouldReceiveShadow={true}
+          overrideColorHex="#ffffff"
+          useLambertWhiteMaterial={true}
+        />
+        <GLBModel
+          modelUrl="/assets/3d-models/shadow_man.glb"
+          modelPosition={[-100, 0, 2]}
+          modelScale={1}
+          overrideColorHex="#ffffff"
+          shouldReceiveShadow={false}
+          forceFlatColorHex="#ffffff"
+        />
+      </Suspense>
+
+      {/* Stand Model built from individual components */}
+      <Suspense fallback={<ModelLoadingFallback />}>
+        <StandBuilder
+          selectedColor={selectedColor}
+          desiredWidth={width}
+          desiredHeight={height}
+          desiredDepth={depth}
+          desiredPlintHeight={currentPlintHeight}
+        />
+      </Suspense>
+    </>
+  )
+})
+
+// Main furniture viewer component
+interface FurnitureViewerProps {
+  selectedColor: string
+  width: number
+  height: number
+  depth: number
+  currentPlintHeight: number
+}
+
+const FurnitureViewerComponent: React.FC<FurnitureViewerProps> = ({
+  width,
+  selectedColor,
+  height,
+  depth,
+  currentPlintHeight,
+}) => {
+  const handleCanvasCreated = useCallback(({ gl: webGlRenderer, scene: threeScene }: { gl: THREE.WebGLRenderer; scene: THREE.Scene }) => {
+    webGlRenderer.outputColorSpace = THREE.SRGBColorSpace
+    webGlRenderer.toneMapping = THREE.ACESFilmicToneMapping
+    webGlRenderer.toneMappingExposure = 1.0
+    webGlRenderer.shadowMap.enabled = true
+    webGlRenderer.shadowMap.type = THREE.PCFSoftShadowMap
+    webGlRenderer.shadowMap.autoUpdate = true
+    threeScene.fog = new THREE.Fog('#f9f9f9', 150, 400)
+  }, [])
+
+  return (
+    <div style={{ width: '100%', height: '100%', minHeight: '500px' }}>
+      <Canvas
+        camera={{
+          position: [-150, 150, 150],
+          fov: 60,
+          near: 0.5,
+          far: 1000,
+        }}
+        shadows
+        dpr={[1, 1.75]}
+        gl={{ powerPreference: 'high-performance' }}
+        style={{
+          background: 'linear-gradient(135deg, #f9f9f9 0%,#f9f9f9 100%)',
+        }}
+        onCreated={handleCanvasCreated}
+      >
+        {/* Orbit controls for rotation and zoom */}
+        <OrbitControls
+          enablePan={false}
+          enableZoom={true}
+          enableRotate={true}
+          enableDamping={false}
+          dampingFactor={0}
+          minDistance={2}
```

Comment:
minDistance={2}
```suggestion
          minDistance={100}
```
looks better